### ISSUE TITLE
[BUG] Modal Close Button

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#541]
**Describe the bug**

Modal Close Button is missing important accessibility attributes

**To Reproduce**
1. Go to https://modus-web-components.trimble.com/?path=/docs/components-modal--default
2. Click on 'open modal'
3. The Close button should have attributes `role="button"` and `aria-label="Close"`

**Screenshots**

![image](https://user-images.githubusercontent.com/1212885/177485377-e90cd170-51af-4ae8-8ea5-77e63c13a36e.png)

There is a z-index issue with this too
